presto-main/src/main/java/io/prestosql/operator/LookupJoinOperator.java

```diff
@@ -336,8 +342,13 @@ private boolean joinCurrentPosition(LookupSource lookupSource, DriverYieldSignal
                     joinSourcePositions++;
                 }
 
-                // get next position on lookup side for this probe row
-                joinPosition = lookupSource.getNextJoinPosition(joinPosition, probe.getPosition(), probe.getPage());
+                if (!matchSingleBuildRow || !currentProbePositionProducedRow) {
```

Comment:
What if we could make pass this flag (matchSingleBuildRow) so when we build the hashTable we could make sure that only one element exists ? Or override `getNextJoinPosition` to return -1. Not sure if it would affect other places

Reply:
Or maybe a simple `PositionalLink` that could return -1 for `PositionalLink#next`

Reply:
That would work when there are no extra filters in `JoinNode#filter`. It's a good idea. Could it be a follow up?